src/overlays/actors/ovl_En_Dnb/z_en_dnb.c

```diff
@@ -22,34 +45,273 @@ const ActorInit En_Dnb_InitVars = {
     (ActorFunc)EnDnb_Draw,
 };
 
-#endif
+void func_80A4FDD0(EnDnbParticle* particle, EnDnb* this, s16* alloc, s32 idx) {
+    Vec3f sp1C;
+    s32 idx2 = idx * 3;
+
+    sp1C.x = alloc[idx2 + 0] + this->dyna.actor.world.pos.x;
+    sp1C.y = alloc[idx2 + 1] + this->dyna.actor.world.pos.y;
+    sp1C.z = alloc[idx2 + 2] + this->dyna.actor.world.pos.z;
+    particle->unk_00 = sp1C;
+    particle->unk_0C = sp1C;
+    particle->unk_24 = Math_Vec3f_Yaw(&this->dyna.actor.world.pos, &sp1C);
+    particle->unk_18 = D_801D15BC;
+}
+
+s32 func_80A4FEBC(EnDnbParticle* particle, f32 arg1) {
+    s32 ret = false;
+
+    if ((DECR(particle->unk_26) == 0) && (arg1 < particle->unk_0C.y)) {
+        Math_ApproachF(&particle->unk_30, 1.0f, 0.4f, 1.0f);
+        particle->unk_2C += particle->unk_34;
+        particle->unk_0C.x += particle->unk_30 * Math_SinS(particle->unk_24);
+        particle->unk_0C.z += particle->unk_30 * Math_CosS(particle->unk_24);
+        particle->unk_0C.y += particle->unk_2C;
+        if (particle->unk_0C.y <= arg1) {
+            particle->unk_0C.y = arg1;
+        }
+        particle->unk_18.x += particle->unk_1E.x;
+        particle->unk_18.y += particle->unk_1E.y;
+        particle->unk_18.z += particle->unk_1E.z;
+        ret = true;
+    }
+
+    return ret;
+}
+
+void func_80A4FFE8(EnDnbParticle* particle, s16 arg1) {
+    particle->unk_0C = particle->unk_00;
+    particle->unk_1E.x = (Rand_ZeroOne() - 0.5f) * 400.0f;
+    particle->unk_1E.y = (Rand_ZeroOne() - 0.5f) * 400.0f;
+    particle->unk_1E.z = (Rand_ZeroOne() - 0.5f) * 400.0f;
+    particle->unk_18 = D_801D15BC;
+    particle->unk_30 = 40.0f;
+    particle->unk_2C = 0.0f;
+    particle->unk_26 = arg1;
+    particle->unk_34 = (Rand_ZeroOne() * -2.0f) - 2.0f;
+}
+
+s32 func_80A500F8(EnDnb* this) {
+    static Vec3f D_80A50CB0 = { 0.0f, 0.0f, 1000.0f };
+
```

Comment:
```suggestion
```

Reply:
I can't see the change here, which line am I removing? I thought there's always a gap between static data and variables, and then another before the code.

Reply:
No, point is there is usually no gap between the in-function static and the variables.